crates/l2/contracts/src/l1/CommonBridge.sol

```diff
@@ -45,6 +47,18 @@ contract CommonBridge is
     /// @dev It is used as the nonce of the mint transaction created by the L1Watcher.
     uint256 public depositId;
 
+    /// @notice Address of the bridge on the L2
+    /// @dev It's used to validate withdrawals
+    address public constant L2_BRIDGE_ADDRESS = address(0xffff);
+
+    /// @notice How much of each L1 token was deposited to each L2 token.
+    /// @dev Stored as L1 -> L2 -> amount
+    /// @dev Prevents L2 tokens from faking their L1 address and stealing tokens
+    mapping (address => mapping (address => uint256)) public depositsERC20;
```

Comment:
We don't need to store the amount here, we can just call the `balanceOf(address)` method of the L1 contract (or `address(this).balance` in case of ETH). Furthermore, do we need to check the balance at all? If the bridge doesn't have enough funds the transfer will just fail and so the withdrawal

Reply:
We need to track the amount of tokens deposited into `<L2 token>` from `<L1 token>` to stop people from creating some random `<L2 token>`, depositing a minimum amount of an existing `<L1 token>` into it, and then withdrawing all of the bridge's balance of `<L1 token>`.

We should probably add this exact case in the docs 🤔